syntax = "proto3";

message Wish{
  string id = 1;
}

message Lib{
  string id = 1;
}

message GameLibrary {
  string url = 1;
  string types = 2;
  string name = 3;
  string desc_snippet = 4;
  string recent_reviews = 5;
  string all_reviews = 6;
  string release_date = 7;
  string developer = 8;
  string publisher = 9;
  string popular_tags = 10;
  string game_details = 11;
  string languages = 12;
  string achievements = 13;
  string genre = 14;
  string game_description = 15;
  string mature_content = 16;
  string minimum_requirements = 17;
  string recommended_requirements = 18;
  string original_price = 19;
  string discount_price = 20;
  string _id = 21;

}


message ListGamesWishResponse {
  repeated GameLibrary games = 1;
}

message ListGamesWishRequest {
  string token = 1;
}

message AddGameWishResponse {
  string message = 1;
}

message AddGameWishRequest {
  string id = 1;
  string token = 2;
}

message DeleteGameWishRequest {
  string id = 1;
  string token = 2;
}

message DeleteGameWishResponse {
  string message = 1;
}


service Wishlist {

  // Returns a list of games in the wish
  rpc ListGames(ListGamesWishRequest) returns (ListGamesWishResponse);

  // Add a new game to the wish
  rpc AddGame(AddGameWishRequest) returns (AddGameWishResponse);

  // Deletes a game from the wish
  rpc DeleteGame(DeleteGameWishRequest) returns (DeleteGameWishResponse);
}
